### PR TITLE
Fixes #31247 - deleting wrong repo reference on cv delete

### PR DESCRIPTION
**Pull request by jlsherrill**
previously the wrong repo reference object was deleted

### CODE REVIEW

File: test/actions/pulp3/content_view/delete_repository_references_test.rb [hunk 1]
```diff
@@ -13,13 +13,15 @@ def setup
...
+      repo_reference = Katello::Pulp3::RepositoryReference.find_by(:content_view => @content_view, :root_repository_id => @repo.root.id)
+      library_repo_ref = Katello::Pulp3::RepositoryReference.create!(:root_repository_id => @repo.root.id, :content_view_id => @repo.library_instance.content_view.id, :repository_href => '/some/fake/href')
+
       assert repo_reference
       assert Katello::Pulp3::Api::File.new(@primary).repositories_api.read(repo_reference.repository_href)
 
       ForemanTasks.sync_task(::Actions::Pulp3::ContentView::DeleteRepositoryReferences, @content_view, @primary)
       refute Katello::Pulp3::RepositoryReference.find_by(:id => repo_reference.id)
-
+      assert Katello::Pulp3::RepositoryReference.find_by(:id => library_repo_ref.id)
```
Comment (jlsherrill) [inline test/actions/pulp3/content_view/delete_repository_references_test.rb:24]: note that this test reproduces the error on master

**Review by ianballou — COMMENTED** [review]
This needs an update to get the tests passing: https://github.com/Katello/katello/blob/master/test/support/pulp3_support.rb#L22

**Review by ianballou — COMMENTED** [review]
It's working fine for me. I'll ACK once the tests pass.

**Comment by ianballou** [comment]
Rubocop error and a VCR issue is remaining.  Other issues are the exporter failures.

File: test/services/katello/pulp3/repository_list_test.rb [hunk 2]
```diff
@@ -23,16 +23,14 @@ def test_list_with_pagination
           User.current = users(:admin)
           @primary = FactoryBot.create(:smart_proxy, :default_smart_proxy, :with_pulp3)
 
-          repo1 = katello_repositories(:pulp3_file_1)
+          repo1 = katello_repositories(:generic_file)
```
Comment (jlsherrill) [inline test/services/katello/pulp3/repository_list_test.rb:26]: This test wasn't really written properly, notice its creating the same repo in pulp3 twice.

**Review by ianballou — APPROVED** [review]
Looks good to me!

**Pull request merged**